exercises/practice/rest-api/.meta/src/RestApi.example.elm

```diff
@@ -0,0 +1,187 @@
+module RestApi exposing (buildDatabase, get, post)
+
+import Dict exposing (Dict)
+import Json.Decode as Decode exposing (Decoder, Error)
+import Json.Encode as Encode exposing (Value)
+
+
+type alias Name =
+    String
+
+
+type alias User =
+    { name : Name
+    , owes : Dict Name Int
+    , owedBy : Dict Name Int
+    , balance : Int
+    }
+
+
+type alias IOU =
+    { lender : Name, borrower : Name, amount : Int }
+
+
+type alias Database =
+    Dict Name User
+
+
+type PostPayload
+    = NewUser User
+    | NewIOU IOU
+
+
+buildDatabase : String -> Result Error Database
+buildDatabase =
+    Decode.decodeString databaseD
```

Comment:
I would probably call this databaseDecoder, but feels like a personal preference type of thing.